File: Accounting/src/main/java/com/experiment/accounting/utils/AuthUtils.java
package com.experiment.accounting.utils;

import com.experiment.accounting.proxy.authorize.AuthorizeProxy;
import com.experiment.accounting.proxy.authorize.model.User;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.oauth2.provider.OAuth2Authentication;
import org.springframework.stereotype.Service;

@Service
public class AuthUtils {

    public static User getAuthenticatedUser(){
        OAuth2Authentication authentication = (OAuth2Authentication) SecurityContextHolder.getContext().getAuthentication();
        return BeanUtil.getBean(AuthorizeProxy.class).getUserByUsername(authentication.getUserAuthentication().getName());
    }
}

File: Inventory/src/main/kotlin/com/experiment/inventory/utils/AuthUtils.kt
package com.experiment.inventory.utils

import com.experiment.inventory.proxy.authorize.AuthorizeProxy
import com.experiment.inventory.proxy.authorize.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.provider.OAuth2Authentication
import org.springframework.stereotype.Service

@Service
class AuthUtils {

    companion object{
        fun getAuthenticatedUser(): User {
            val authentication = SecurityContextHolder.getContext().authentication as OAuth2Authentication
            return BeanUtil.getBean(AuthorizeProxy::class.java)?.getUserByUsername(authentication.userAuthentication.name) ?: User()
        }
    }

}